MaskedSelect.lua

```diff
@@ -0,0 +1,41 @@
+local MaskedSelect, parent = torch.class('nnd.MaskedSelect', 'nn.Module')
+
+--[[ Sets the provided mask value for the module. ]]
+function MaskedSelect:__init()
+  -- mask must be a tensor, convert to byte tensor
+  parent.__init(self)
+end
+
+--[[ Performs maskedSelect operation. ]]
+function MaskedSelect:updateOutput(input)
+  local input, mask = unpack(input)
+  self.output = input:maskedSelect(mask)
+  return self.output
+end
+
+--[[ Reverse maps unmasked gradOutput back to gradInput. ]]
+function MaskedSelect:updateGradInput(input, gradOutput)
+  local input, mask = unpack(input)
+  local maskIndices =
+    torch.range(1, mask:nElement()):resizeAs(mask:double()):maskedSelect(mask)
```

Comment:
Same here. Keep `maskIndices` around as a private member of the class and call `self._maskIndices:range(...)`.

You should also avoid casting the mask to double, so instead of `resizeAs(mask:double())` do `resize(mask:size())`.